### PR TITLE
Interpolator with namelist

### LINKED ISSUE #569: Namelist switch to select IO version for use in libFMS modules
**Is your feature request related to a problem? Please describe.**
fms2_io and mpp_io both exist in the FMS repository.  There is no way to test each version of IO side-by-side in the different modules.

**Describe the solution you'd like**
Add a namelist option to the different modules in libFMS
- [x] diag_manager @thomas-robinson 
- [x] xgrid @thomas-robinson 
- [x] data_override @GFDL-Eric 
- [x] interpolator @uramirez8707 
- [x] amip_interp @uramirez8707 
- [x] topography @rem1776 
- [x] travis yaml and autotools build @rem1776 

**Describe alternatives you've considered**
The current alternative is a compile time switch which is an all-or-nothing solution

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**

1.  Adds the use_mpp_io namelist flag
2. Adds mpp_io's version of `interpolator_init`, `read_data`, `read_data`
3. Moves the part of the code that writes the namelist/version_number to the log file, so that it only writes it once (#558 )

Fixes #569 #558 

**How Has This Been Tested?**
Travis CI
Passes `make check` on skylake with intel19
AM4.0 with intel18 in debug 

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: interpolator/interpolator.F90 [hunk 1]
```diff
@@ -3885,3841 +4125,845 @@ subroutine interp_linear ( grdin, grdout, datin, datout )
...
-logical :: retain_cm3_bug = .true.               !< No description
-
-namelist /interpolator_nml/    &
-                             read_all_on_init, verbose, conservative_interp, retain_cm3_bug
-
-contains
+do i = 1, ndim
+  call mpp_get_atts(axes(i), name=name,len=len,units=units,  &
+                    calendar=file_calendar, sense=sense)
+  select case(name)
```
Comment (wrongkindofdoctor) [inline interpolator/interpolator.F90:4221]: Making this `case(trim(name))` would be safer if possible.

Comment (uramirez8707) [reply interpolator/interpolator.F90:4221]: Good idea. Thank you.

**Review by wrongkindofdoctor — APPROVED** [review]

File: interpolator/interpolator.F90 [hunk 2]
```diff
@@ -479,6 +509,62 @@ subroutine interpolator_init( clim_type, file_name, lonb_mod, latb_mod, &
...
+!--------------------------------------------------------------------
+! namelist input
+!--------------------------------------------------------------------
+
+the_file_exists = fms2_io_file_exist('input.nml')
+
+if (the_file_exists) then
+    read (input_nml_file, nml=interpolator_nml, iostat=io)
+    ierr = check_nml_error(io,'interpolator_nml')
+end if
```
Comment (bensonr) [inline interpolator/interpolator.F90:529]: Should there also be internal namelist file read logic here?

File: interpolator/interpolator.F90 [hunk 3]
```diff
@@ -742,30 +811,30 @@ subroutine interpolator_init( clim_type, file_name, lonb_mod, latb_mod, &
            & trim(adjustl(lowercase(file_calendar))) == 'julian')  .or. &
            & (model_calendar == NOLEAP .and.   &
            & trim(adjustl(lowercase(file_calendar))) == 'noleap') )  then
-        call mpp_error (NOTE, 'interpolator_mod: Model and file&
+        call mpp_error (NOTE, 'interpolator[2]_mod: Model and file&
```
Comment (bensonr) [inline interpolator/interpolator.F90:814]: Why the change to the module name in the error messages?  The module is still interpolator_mod.  I find the '[2]' a distraction.

**Review by bensonr — CHANGES_REQUESTED** [review]
There's a lot of code that is different here and difficult to determine if everything has been updated correctly.

File: interpolator/interpolator.F90 (hunk 2 above)
Comment (uramirez8707) [reply interpolator/interpolator.F90:529]: I thought we were working on getting rid of the `#ifdef INTERNAL_FILE_NML` because it was **_always_** set to true?

Comment (bensonr) [reply interpolator/interpolator.F90:529]: You're right.  I misread the code.

**Review by bensonr — APPROVED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]
